### PR TITLE
Add one smoke test for set a resource tag case

### PR DESCRIPTION
**Pull request by Grayer123**
## Description

Add one smoke test for set a resource tag case
[Address issue: Add end-to-end test automation verifying case preservation in Tag keys and Values](https://github.com/Azure/azure-powershell/issues/7827) 

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Resources/Resources.Test/ScenarioTests/ResourceTests.ps1 [hunk 1]
```diff
@@ -856,4 +856,51 @@ function Test-RemoveASetOfResources
...
+    $apiversion = "2014-04-01"
+    $resourceType = "Providers.Test/statefulResources"
+ 
+    try
+    {
+        # Test
+        New-AzResourceGroup -Name $rgname -Location $rglocation
+        $resource = New-AzResource -Name $rname -Location $rglocation -Tags @{testtag = "testval"} -ResourceGroupName $rgname -ResourceType $resourceType -PropertyObject @{"key" = "value"} -SkuObject @{ Name = "A0" } -ApiVersion $apiversion -Force
+ 
+        # Verify tags and their casing
```
Comment (hamidtirmizi) [inline src/Resources/Resources.Test/ScenarioTests/ResourceTests.ps1]: Need to implement this

File: src/Resources/Resources.Test/ScenarioTests/ResourceTests.ps1 [hunk 2]
```diff
@@ -856,4 +856,51 @@ function Test-RemoveASetOfResources
...
+        # resource.Tags key is exactly "testtag" with case sensitive match
+ 
+        # Set resource (add a new tag with key TestTag2 = "TestVal2")
+        Set-AzResource -Tags @{testtag = "testval"; TestTag2 = "TestVal2"} -ResourceGroupName $rgname -ResourceName $rname -ResourceType $resourceType -Properties @{"key2" = "value2"} -Force
+        $resource = Get-AzResource -ResourceGroupName $rgname -ResourceName $rname -ResourceType $resourceType
+ 
+        # Verify tags and their casing (note upper case in the key)
```
Comment (hamidtirmizi) [inline src/Resources/Resources.Test/ScenarioTests/ResourceTests.ps1]: Need to implement this

File: src/Resources/Resources.Test/ScenarioTests/ResourceTests.ps1 [hunk 3]
```diff
@@ -856,4 +856,51 @@ function Test-RemoveASetOfResources
...
+        # resource.Tags keys are exactly "testtag" and "TestTag2" with case sensitive match
+        # case sensitive match for keys too
+ 
+        # Set resource (add a new tag with key TestTag2 = "TestVal2")
+        Set-AzResource -Tags @{Testtag = "testVAL"; TestTag2 = "TestVal2"} -ResourceGroupName $rgname -ResourceName $rname -ResourceType $resourceType -Properties @{"key2" = "value2"} -Force
+        $resource = Get-AzResource -ResourceGroupName $rgname -ResourceName $rname -ResourceType $resourceType
+ 
+        # Verify tags and their casing (note upper case in the key)
```
Comment (hamidtirmizi) [inline src/Resources/Resources.Test/ScenarioTests/ResourceTests.ps1]: Need to implement this

**Comment by msJinLei** [comment]
@hamidtirmizi Could you take a look at the @Grayer123's change? Thanks

**Comment by msJinLei** [comment]
@Grayer123 Please resolve the conflict

**Comment by Grayer123** [comment]
Close this pr due to a force-pushed into master, as a [new pr](https://github.com/Azure/azure-powershell/pull/11623) created.

**Pull request closed without merging**